tests/e2e/util.go

```diff
@@ -6331,142 +6331,161 @@ func cleaupStatefulset(client clientset.Interface, ctx context.Context, namespac
-	podSpec.Spec.Subdomain = "sample-pe-svc"
-	var x int64 = 300
-	var tolerations []v1.Toleration
-	for _, taintkey := range taintKeys {
-		var toleration v1.Toleration
-		toleration.Key = taintkey
-		toleration.Operator = v1.TolerationOpExists
-		toleration.TolerationSeconds = &x
-		toleration.Effect = v1.TaintEffectNoExecute
-		framework.Logf("toleration: %v", toleration)
-		tolerations = append(tolerations, toleration)
-	}
-	podSpec.Spec.Tolerations = tolerations
-	framework.Logf("pod spec: %v", podSpec)
-	return podSpec
+	cmd = "kubectl apply -f pvc.yaml"
+	framework.Logf("Invoking command '%v' on host %v", cmd,
+		svcMasterIP)
+	cmdResult, err = sshExec(sshClientConfig, svcMasterIP,
+		cmd)
+	if err != nil || cmdResult.Code != 0 {
+		fssh.LogResult(cmdResult)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			cmd, svcMasterIP, err)
+	}
+	return nil
 }
 
-// createVsanDPvcAndPod creates pvc and pod for vsan direct as per wffc policy.
-// It ensures that pvc and pod is in in healthy state.
-func createVsanDPvcAndPod(client clientset.Interface, ctx context.Context,
-	namespace string, sc *storagev1.StorageClass, pvcName string,
-	podName string) (*v1.PersistentVolumeClaim, *v1.Pod) {
-	framework.Logf("Creating pvc %s with storage class %s", pvcName, sc.Name)
-	pvclaim, err := fpv.CreatePVC(client, namespace,
-		getVsanDPersistentVolumeClaimSpecWithStorageClass(namespace,
-			diskSize, sc, pvcName, podName, ""))
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
-
-	ginkgo.By("Expect claim status to be in Pending state")
-	err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimPending, client,
-		namespace, pvclaim.Name, framework.Poll, time.Minute)
-	gomega.Expect(err).NotTo(gomega.HaveOccurred(),
-		"Failed to find the volume: %s in pending state with err: %v", pvcName, err)
+// applyVsanDirectPodYaml creates specific pod spec and applies vsan direct pod yaml on svc master
+func applyVsanDirectPodYaml(sshClientConfig *ssh.ClientConfig, svcMasterIP string, svcNamespace string,
+	pvcName string, podName string) error {
+	framework.Logf("POD yaML")
+	cmd := fmt.Sprintf("sed -i -e '/^metadata:/,/namespace:/{/^\\([[:space:]]*namespace: \\).*/s//\\1%s/}' pod.yaml", svcNamespace)
+	framework.Logf("Invoking command '%v' on host %v", cmd,
+		svcMasterIP)
+	cmdResult, err := sshExec(sshClientConfig, svcMasterIP,
+		cmd)
+	if err != nil || cmdResult.Code != 0 {
+		fssh.LogResult(cmdResult)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			cmd, svcMasterIP, err)
+	}
 
-	ginkgo.By("Creating a pod")
-	podSpec := getVsanDPodSpec(namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, execCommand, podName)
-	pod, err := client.CoreV1().Pods(namespace).Create(ctx, podSpec, metav1.CreateOptions{})
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	cmd = fmt.Sprintf("sed -i -e '/^metadata:/,/claimName:/{/^\\([[:space:]]*claimName: \\).*/s//\\1%s/}' pod.yaml", pvcName)
```

Comment:
What exactly are we doing here in this piece of regex code ?
Any reason why we using pod.yaml and then creating pod ?